### DISCUSSION TITLE
How to know if neovim is focused when opened?

### DISCUSSION THREAD

**Discussion by fent** [neovim/neovim#29911] [Q&A]
I'm using the `FocusGained` and `FocusLost` events to keep track if neovim is currently focused. I'm using tmux and "focused" in this case means the tmux pane that neovim is on is focused.

However, these events don't fire at the start. If I open neovim by running `sleep 5 && nvim`, and unfocus the pane before 5s, neovim opens without having focus. How can I know if this has happened?

**Comment by zeertzjq** [comment]
According to https://unix.stackexchange.com/a/480138 this is only possible in some terminal emulators (e.g. GNOME Terminal), in which case `FocusGained`/`FocusLost` are triggered on startup.

**Comment by gpanders** [comment]
[Ghostty](https://mitchellh.com/ghostty) is also considering enabling these initial focus reports, and I've started working on a branch in Neovim to correctly handle them.

The tricky bit is that these events usually indicate a _change_ in focus state, so a `FocusGained` event implies that Neovim did not have focus before, and has since gained focus. So firing a `FocusGained` event on startup when the terminal window already has focus might have unintended side effects for any plugin code that depends on these "edge triggered" semantics.

What I intend to do in my implementation is assume that Nvim has focus on startup. If the initial focus report indicates that Nvim does _not_ have focus, then a `FocusLost` event will be fired; otherwise, the initial focus report is ignored.
